source/server/http/admin.cc

```diff
@@ -411,54 +411,67 @@ Http::Code AdminImpl::handlerStats(absl::string_view url, Http::HeaderMap& respo
                                    Buffer::Instance& response, AdminStream& admin_stream) {
   Http::Code rc = Http::Code::OK;
   const Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
+
+  const bool used_only = !(params.find("usedonly") == params.end());
+  const bool has_format = !(params.find("format") == params.end());
+
   std::map<std::string, uint64_t> all_stats;
   for (const Stats::CounterSharedPtr& counter : server_.stats().counters()) {
-    all_stats.emplace(counter->name(), counter->value());
+    const bool add_counter = used_only ? counter->used() : true;
+    if (add_counter) {
```

Comment:
What you have is correct, though the logic could be simpler a bit.  Invert the bools above, e.g.
```
const bool show_unused = params.find("used_only") == params.end();
```

And then you can skip the 'add_counter', 'add_gauge', and 'add_histogram' temps and ternary expressions, and just write:
```
if (show_unused || counter->used()) {
```